### PR TITLE
Add `responseErrorFromObject` to create detailed error using response object

### LINKED ISSUE #15: Parse response body even if the response status code is not 2xx
Currently if the response status code is not 2xx, it won't parse the response body at all.

However sometimes the response body may contain useful info even if the response status code is not 2xx (mostly 4xx), for example, validation errors of object creation.

Can we please at least having an option to choose whether to parse the response body even if the response status code is not 2xx?

### PR DESCRIPTION
**Pull request by ishkawa**
Resolves #15.

### CODE REVIEW

File: APIKit/APIKit.swift [hunk 1]
```diff
@@ -161,10 +165,15 @@ public class API: NSObject, NSURLSessionDelegate, NSURLSessionDataDelegate {
                 }
                 
                 let statusCode = (URLResponse as? NSHTTPURLResponse)?.statusCode ?? 0
-                if !contains(200..<300, statusCode) {
-                    let userInfo = [NSLocalizedDescriptionKey: "received status code that represents error"]
-                    let error = NSError(domain: APIKitErrorDomain, code: statusCode, userInfo: userInfo)
-                    dispatch_async(mainQueue, { handler(.Failure(Box(error))) })
+                if !contains(self.acceptableStatusCodeRange, statusCode) {
+                    var error: NSError = {
```
Comment (ikesyo) [inline APIKit/APIKit.swift]: Style: `let` can be used here.

File: APIKit/APIKit.swift [hunk 2]
```diff
@@ -115,6 +115,10 @@ public class API: NSObject, NSURLSessionDelegate, NSURLSessionDataDelegate {
         return instancePair.1
     }
 
+    public class var acceptableStatusCodeRange: Range<Int> {
```
Comment (ikesyo) [inline APIKit/APIKit.swift]: Why don't you use `[Int]` or `NSSet`, `NSIndexSet` (or `Set<Int> in Swift 1.2`) instead? This `Range` type makes it impossible to use status codes with interval.

**Comment by ikesyo** [comment]
Looks good :sparkles:

**Comment by ishkawa** [comment]
Thanks for reviewing carefully! I have fixed code you pointed out and will merge this into develop/0.4.

**Pull request merged**